plugins/backfill/main.py

```diff
@@ -0,0 +1,162 @@
+# -*- coding: utf-8 -*-
+
+# Inbuilt Imports
+import os
+import json
+import logging
+import datetime
+import re
+
+# Custom Imports
+import flask
+from flask import request
+from flask_admin import BaseView, expose
+from flask_appbuilder import expose as app_builder_expose, BaseView as AppBuilderBaseView,has_access
+from airflow import configuration
+
+from shelljob import proc
+
+# Inspired from
+# https://mortoray.com/2014/03/04/http-streaming-of-command-output-in-python-flask/
+# https://www.endpoint.com/blog/2015/01/28/getting-realtime-output-using-python
+# RBAC inspired from 
+# https://github.com/teamclairvoyant/airflow-rest-api-plugin
+
+
+# Set your Airflow home path
+airflow_home_path = os.environ['AIRFLOW_HOME']
+
+# Local file where history will be stored
+FILE = airflow_home_path + '/logs/backfill_history.txt'
+
+rbac_authentication_enabled = configuration.getboolean("webserver", "RBAC")
+
+# RE for remove ansi escape characters
+ansi_escape = re.compile(r'\x1B[@-_][0-?]*[ -/]*[@-~]')
+
+# Creating a flask admin BaseView
+def file_ops(mode, data=None):
+    """ File operators - logging/writing and reading user request """
+    if mode == 'r':
+        try:
+            with open(FILE, 'r') as f:
+                return f.read()
+        except IOError:
+            with open(FILE, 'w') as f:
+                return f.close()
+
+    elif mode == 'w' and data:
+        today = datetime.datetime.now()
+        print(os.getcwd())
+        with open(FILE, 'a+') as f:
+            file_data = '{},{}\n'.format(data, today)
+            f.write(file_data)
+            return 1
+
+def get_baseview():
+    if rbac_authentication_enabled == True:
+        return AppBuilderBaseView
+    else:
+        return BaseView
+
+class Backfill(get_baseview()):
+
+    route_base = "/admin/backfill/"
+
+    if rbac_authentication_enabled == True:
+        @app_builder_expose('/')
+        def list(self):
+            """ Render the backfill page to client with RBAC"""
+            return self.render_template("backfill_page.html",
+                                        rbac_authentication_enabled=rbac_authentication_enabled)        
+    else:
+        @expose('/')
+        def base(self):
+            """ Render the backfill page to client """
+            return self.render("backfill_page.html")
+
+    @expose('/stream')
+    @app_builder_expose('/stream')
+    def stream(self):
+        """ Runs user request and outputs console stream to client"""
+        dag_name = request.args.get("dag_name")
+        start_date = request.args.get("start_date")
+        end_date = request.args.get("end_date")
+        clear = request.args.get("clear")
+        task_regex = request.args.get("task_regex")
+        use_task_regex = request.args.get("use_task_regex")
+
+        # TODO - should we use -c as an argument to skip confirmation on the clearing?
```

Comment:
will deploy this, and further changes to the plugin can be made without changes to the container.